### ISSUE TITLE
[Question] Effect types

### ISSUE THREAD

**Issue by LossyDragon** [libxmp/libxmp#321]
This is more of a question to see if libxmp has a feature that I'm not seeing or looking in the wrong direction for a desired feature. 

I'm looking for a way to output a alphanumeric value of an effect type (IE: Arpeggio  for .it would be _J_xy while a .mod module is _0_xy, etc.)

I know I can obtain the value from xmp_event -> fxt or f2t per frame which I'm referencing effects.h as a proof of concept to lookup the value and tie it to a corresponding character. But I noticed that some defines in effects.h have the same value and some of the loader's do not reference this list of effects. 

I thought about making some type of look up table for some of the common formats, but this might exclude other formats xmp support for a robust pattern viewer.

**Comment by AliceLR** [comment]
There's nothing like this in libxmp currently (that I'm aware of). Due to the way libxmp translates effects to a common internal representation, getting an appropriate symbol for each effect code might not always be straightforward and may need to be handled for every individual format. @cmatsuoka might have a better idea of how to handle this.

re: the loaders that don't reference effects symbols or have an `fx` array, those usually map effect codes the same as Protracker.

**Comment by LossyDragon** [comment]
Thanks for the update! I managed to expand on the initial idea and tied almost most of the loaders with an `fx` array with my own. Tying based on the mod type (name), while falling back to protracker. 

Bit ugly and far from perfect, but it seems to be doing the job with the pile of mods I used to test. [Effects.kt](https://github.com/LossyDragon/xmp-android/blob/kotlin/app/src/main/java/org/helllabs/android/xmp/player/viewer/Effects.kt)

**Comment by NoSuck** [comment]
> ... may need to be handled for every individual format.

Yes.  Also, such conversion can be ambiguous (or irreversible, if from the other direction).  The IT format's S and Z effects map to numerous libxmp internal values for `fxt`, for example.
